P2PEnabledConference.js

```diff
@@ -0,0 +1,899 @@
+
+    /**
+     * Local role change may trigger new P2P session if 'everyone's a moderator'
+     * plugin is enabled.
+     * @inheritDoc
+     * @override
+     */
+    onLocalRoleChanged (newRole) {
+        super.onLocalRoleChanged(newRole);
+        // Maybe start P2P
+        this._startStopP2PSession();
+    }
+
+    /**
+     * @inheritDoc
+     * @override
+     */
+    onMemberJoined (jid, nick, role, isHidden) {
+        super.onMemberJoined(jid, nick, role, isHidden);
+
+        this._startStopP2PSession();
+    }
+
+    /**
+     * @inheritDoc
+     * @override
+     */
+    onMemberLeft (jid) {
+        super.onMemberLeft(jid);
+
+        this._startStopP2PSession(true /* triggered by user left event */);
+    }
+
+    /**
+     * Called when {@link XMPPEvents.CONNECTION_INTERRUPTED} occurs on the P2P
+     * connection.
+     */
+    onP2PIceConnectionInterrupted () {
+        this.isP2PConnectionInterrupted = true;
+        if (this.p2pEstablished)
+            this._originalEmit(JitsiConferenceEvents.CONNECTION_INTERRUPTED);
+    }
+
+    /**
+     * Called when {@link XMPPEvents.CONNECTION_RESTORED} occurs on the P2P
+     * connection.
+     */
+    onP2PIceConnectionRestored () {
+        this.isP2PConnectionInterrupted = false;
+        if (this.p2pEstablished)
+            this._originalEmit(JitsiConferenceEvents.CONNECTION_RESTORED);
+    }
+
+    /**
+     * @inheritDoc
+     * @override
+     */
+    onRemoteTrackAdded (track) {
+        if (track.isP2P && !this.p2pEstablished) {
+            logger.info(
+                "Trying to add remote P2P track, when not in P2P - IGNORED");
+        } else if (!track.isP2P && this.p2pEstablished) {
+            logger.info(
+                "Trying to add remote JVB track, when in P2P - IGNORED");
+        } else {
+            super.onRemoteTrackAdded(track);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     * @override
+     */
+    onTransportInfo (jingleSession, transportInfo) {
+        if (this.peerToPeerSession === jingleSession) {
+            logger.info("Doing set transport-info");
+            this.peerToPeerSession.addIceCandidates(transportInfo);
+        }
+    }
+
+    /**
+     * Manually starts new P2P session (should be used only in the tests).
+     */
+    startPeer2PeerSession() {
+        const peers = this.getParticipants();
+        // Start peer to peer session
+        if (peers.length > 0) {
+            const peerJid = peers[0].getJid();
+            this._startPeer2PeerSession(peerJid);
+        } else {
+            logger.error("No participant to start the P2P session with !");
+        }
+    }
+
+    /**
+     * Manually stops the current P2P session (should be used only in the tests)
+     */
+    stopPeer2PeerSession() {
+        this._stopPeer2PeerSession();
+    }
+}
+
+/**
+ * This is a fake {@link ChatRoom} passed to the P2P {@link JingleSessionPC}
+ * in order to capture events emitted on it's event emitter (Jingle session uses
```

Comment:
Jingle session was intentional just like chat room was not ChatRoom, so I will keep it this way if it's not a big deal for you. Or I will have to change both.